service/service.go

```
@@ -397,8 +397,19 @@ type RunningDriver struct {
 }
 
 func (r *RunningDriver) Start() (string, error) {
-	port := rand.Intn(1000) + 10*1000 // KontainerDriver port range 10,000 - 11,000
-	listenAddress := fmt.Sprintf("%v%v", ListenAddress, port)
+	ephemeralListenAddress := fmt.Sprintf("%s0", ListenAddress)
+	p, err := net.Listen("tcp", ephemeralListenAddress) // passing this port will cause go to provide open ephemeral port
+	if err != nil {
+		return "", err
+	}
+
+	listenAddress := p.Addr().String()
+	p.Close()
```

Why is the error being ignored?